include/lemlib/motions/differential/differentialMotion.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "units/Pose.hpp"
+#include "lemlib/chassis/differential/differentialSteerRequest.hpp"
+
+namespace lemlib {
+
+/**
+ * @class DifferentialMotion
+ *
+ * @brief Abstract class which represents a motion algorithm for a differential drive robot
+ *
+ * We use this abstraction to allow for different motion algorithms to be used with the same interface.
+ * This allows us to easily create and maintain motion algorithms while also allowing the user to
+ * make custom motion algorithms without having to recompile the library.
+ */
+class DifferentialMotion {
```

Comment:
What does the life cycle for this look like? Is the motion object intended to be created when starting the motion and then destroyed once complete, or can it be reused?

Reply:
It is created once the motion starts, and is destroyed when completed.

Reply:
I'll specify this in its documentation